File: src/main/java/com/thunderbear06/ai/task/Task.java
package com.thunderbear06.ai.task;

import com.thunderbear06.entity.android.AndroidEntity;

public abstract class Task {
    protected final AndroidEntity android;

    protected Task(AndroidEntity android) {
        this.android = android;
    }

    public abstract String getName();
    public abstract boolean shouldTick();
    public abstract void firstTick();
    public abstract void tick();
    public abstract void lastTick();
}

File: src/main/kotlin/com/thunderbear06/ai/task/Task.kt
package com.thunderbear06.ai.task

import com.thunderbear06.entity.android.AndroidEntity

abstract class Task(@JvmField protected val android: AndroidEntity) {
    abstract val name: String
    abstract fun shouldTick(): Boolean
    abstract fun firstTick()
    abstract fun tick()
    abstract fun lastTick()
}
